server/client_server.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	bitbucketv1 "github.com/gfleury/go-bitbucket-v1"
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+)
+
+// TODO: This will be changed to create the main structs when modularized
+type Link struct {
+	Href string `json:"href"`
+}
+
+type BitbucketUser struct {
+	AccountId int    `json:"id"`
+	Username  string `json:"name"`
+	Links     struct {
+		Self []Link `json:"self"`
+	} `json:"links"`
+}
+
+type BitbucketServerClient struct {
+	apiClient *bitbucketv1.APIClient
+
+	selfHostedURL    string
+	selfHostedAPIURL string
+
+	token oauth2.Token
+}
+
+func NewClientServer() *BitbucketServerClient {
+	return &BitbucketServerClient{}
+}
+
+// TODO: This method needs to be changed when Modularization is built
+func (c *BitbucketServerClient) Connect(config Configuration, token oauth2.Token, ts oauth2.TokenSource) *BitbucketServerClient {
+	// setup Oauth context
+	auth := context.WithValue(context.Background(), bitbucketv1.ContextOAuth2, ts)
+
+	tc := oauth2.NewClient(auth, ts)
+
+	// create config for bitbucket API
+	configBb := bitbucketv1.NewConfiguration(config.BitbucketAPISelfHostedURL)
+	configBb.HTTPClient = tc
+
+	c.apiClient = bitbucketv1.NewAPIClient(context.Background(), configBb)
+	c.selfHostedURL = config.BitbucketSelfHostedURL
+	c.selfHostedAPIURL = config.BitbucketAPISelfHostedURL
+	c.token = token
+
+	return c
+}
+
+func (c *BitbucketServerClient) getWhoAmI() (string, error) {
+	baseUrl := fmt.Sprintf("%s/plugins/servlet/applinks/whoami", c.selfHostedURL)
```

Comment:
Maybe call this `requestURL` since this is the full URL for the http request

```suggestion
	requestURL := fmt.Sprintf("%s/plugins/servlet/applinks/whoami", c.selfHostedURL)
```

Also due to Go convention there is a enabled linting rule to capitalize these values:
- `Url` as `URL`
- `Id` as `ID`

I think it's a bit silly but it's a mostly followed standard in Go projects

https://github.com/mattermost/mattermost-plugin-bitbucket/actions/runs/6825212858/job/18731602802?pr=128
```
Error: server/client_server.go:21:2: var-naming: struct field AccountId should be AccountID (revive)
	AccountId int    `json:"id"`
	^
Error: server/client_server.go:61:2: var-naming: var baseUrl should be baseURL (revive)
	baseUrl := fmt.Sprintf("%s/plugins/servlet/applinks/whoami", c.selfHostedURL)
	^
make: *** [Makefile:53: check-style] Error 1
Error: Process completed with exit code 2.
```

Reply:
No problem! this is the convention, so it's good, done!